File: backend/src/main/java/kr/codesquad/issuetracker/core/issue/Issue.java
package kr.codesquad.issuetracker.core.issue;

import java.util.List;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import kr.codesquad.issuetracker.common.BaseTimeEntity;
import kr.codesquad.issuetracker.core.label.Label;
import kr.codesquad.issuetracker.core.milestone.Milestone;

@Entity
public class Issue extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    @Column(name = "open_yn")
    private boolean isOpen;
    @ManyToOne
    private Milestone milestone;
    @OneToMany(mappedBy = "issue")
    private List<Label> labels;
    @OneToMany(mappedBy = "issue")
    private List<Assignee> assignees;
    @OneToMany(mappedBy = "issue")
    private List<Comment> comments;
}

File: IssueTracker/IssueTracker/Entity/Issue.swift
//
//  Issue.swift
//  IssueTracker
//
//  Created by Bibi on 2022/06/16.
//

import Foundation

struct Issue: Codable {
    let title: String
    let body: String?
    let state: String
    let labels: [Label]
    let milestone: Milestone?
    let repository: Repository
}

struct Repository: Codable {
    let name: String
    let owner: Owner
}

struct Owner: Codable {
    let login: String
}
